webapp/packages/plugin-data-viewer/src/DatabaseDataModel/Actions/ResultSet/ResultSetConstraintAction.ts

```diff
@@ -240,6 +240,35 @@ export class ResultSetConstraintAction extends DatabaseDataAction<IDatabaseDataO
 
     return currentConstraint.orderAsc ? EOrder.asc : EOrder.desc;
   }
+
+  updateResults(results: IDatabaseResultSet[]): void {
+    if (!this.source.options) {
+      throw new Error('Options must be provided');
+    }
+
+    const nextResult = results[this.resultIndex];
```

Comment:
can be undefined (please check that `results.lentgh === this.source.getResults().length` or just return)